matlab/schemas/+map/OptImageBar.m

```diff
@@ -20,69 +20,61 @@
-                        getOpticalData(getLocalPath(fullfile(path,name))); % time in sec
-                    
-                    disp 'synchronizing...'
-                    % synchronize to stimulus
-                    tuple =  sync(key, photodiode_signal, photodiode_fs);
-                    
-                    % calculate frame times
-                    frame_times = tuple.signal_start_time + ...
-                        tuple.signal_duration*(1:size(Data,1))/size(Data,1);
-                    
-                    % import Sync table
-                    tuple.frame_times = frame_times;
-                    makeTuples(map.Sync,tuple)
+                    [Data, data_fs] = getOpticalData(key); % time in sec
                     
+                    % get frame times
+                    if ~exist(stimulus.Sync & key)
+                        disp 'synchronizing...'
+                        populate(stimulus.Sync,key)
+                    end
+                    frame_times =fetch1(stimulus.Sync & key,'frame_times');
+
                     % get the vessel image
                     disp 'getting the vessels...'
                     k = [];
                     k.session = key.session;
                     k.animal_id = key.animal_id;
                     k.site_number = fetch1(experiment.Scan & key,'site_number');
-                    vesObj = experiment.Scan & k & 'software = "imager" and aim = "vessels"';
+                    vesObj = experiment.Scan - experiment.ScanIgnored & k & 'software = "imager" and aim = "vessels"';
                     if ~isempty(vesObj)
-                        name = fetch1( vesObj ,'filename');
-                        if isempty(strfind(name,'.h5')); name = [name '.h5'];end
+                        names = fetchn( vesObj ,'filename');
+                        name = names{end};
+                        if ~contains(name,'.h5'); name = [name '.h5'];end
                         filename = getLocalPath(fullfile(path,name));
                         vessels = squeeze(mean(getOpticalData(filename)));
                     end
                     
                 case 'scanimage'
                     
-                    % get Optical data
+                   % get Optical data
                     disp 'loading movie...'
-                    reader = preprocess.getGalvoReader(key);
-                    Data = squeeze(mean(reader(:,:,1,:,:),4));
-                    Data = permute(Data,[3 1 2]);
-                    [nslices, data_fs] = fetch1(preprocess.PrepareGalvo & key,'nslices','fps');
+                    if strcmp(setup,'2P4') % mesoscope
+                        path = getLocalPath(fullfile(path, sprintf('%s*.tif', name)));
+                        reader = ne7.scanreader.readscan(path,'int16',1);
+                        Data = permute(squeeze(mean(reader(:,:,:,:,:))),[3 1 2]);
+                        data_fs = reader.fps;
+                        nslices = length(reader.fieldDepths);
```

Comment:
this will give back number of fields: fieldDepths is a list with corresponding depths per field. reader.nScanningDepths should give you number of slices (or reader.nFields if you actually do want the number of fields).